Build fails on GCC 14

```
#######################################################################
# Compiling C source src/ballet/ed25519/fd_curve25519.c to build/native/gcc/obj/ballet/ed25519/fd_curve25519.o
#######################################################################
mkdir -pv build/native/gcc/obj/ballet/ed25519/ && \
gcc -isystem ./opt/include -DFD_LOG_UNCLEAN_EXIT=1 -DFD_HAS_BACKTRACE=0 -DFD_USING_GCC=1 -march=native -mtune=native -Werror -Wall -Wextra -Wpedantic -Wstrict-aliasing=2 -Wconversion -Wdouble-promotion -Wformat-security -Wimplicit-fallthrough -Wimplicit-fallthrough=2 -O3 -ffast-math -fno-associative-math -fno-reciprocal-math -DFD_HAS_OPTIMIZATION=1 -g  -fno-omit-frame-pointer -D_FORTIFY_SOURCE=2 -fPIC -Wl,-z,now -fstack-protector-strong -DFD_HAS_INT128=1 -DFD_HAS_DOUBLE=1 -DFD_HAS_ALLOCA=1 -DFD_HAS_THREADS=1 -DFD_HAS_OPENSSL=1 -DFD_HAS_X86=1 -DFD_HAS_SSE=1 -DFD_IS_X86_64=1 -DFD_HAS_AESNI=1 -D_XOPEN_SOURCE=700 -DFD_HAS_HOSTED=1 -pthread -DFD_HAS_THREADS=1 -DFD_HAS_ATOMIC=1 -mfpmath=sse -falign-functions=32 -mbranch-cost=5 -falign-jumps=32 -falign-labels=32 -falign-loops=32 -DFD_HAS_UCONTEXT=1 -DFD_BUILD_INFO=\"build/native/gcc/info\" -std=c17 -Wstrict-prototypes -DFD_HAS_SECP256K1=1 -DFD_HAS_ZSTD=1 -DFD_HAS_ROCKSDB=1 -DROCKSDB_LITE=1 -c src/ballet/ed25519/fd_curve25519.c -o build/native/gcc/obj/ballet/ed25519/fd_curve25519.o
In file included from src/ballet/ed25519/fd_curve25519.c:7:
src/ballet/ed25519/fd_curve25519_secure.c:100:1: error: conflicting types for ‘fd_ed25519_scalar_mul_base_const_time’; have ‘fd_ed25519_point_t *(fd_ed25519_point_t *, const uchar *)’ {aka ‘struct fd_curve25519_edwards *(struct fd_curve25519_edwards *, const unsigned char *)’}
  100 | fd_ed25519_scalar_mul_base_const_time( fd_ed25519_point_t * r,
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from src/ballet/ed25519/fd_curve25519.c:1:
src/ballet/ed25519/fd_curve25519.h:145:1: note: previous declaration of ‘fd_ed25519_scalar_mul_base_const_time’ with type ‘fd_ed25519_point_t *(fd_ed25519_point_t *, const uchar *)’ {aka ‘struct fd_curve25519_edwards *(struct fd_curve25519_edwards *, const unsigned char *)’}
  145 | fd_ed25519_scalar_mul_base_const_time( fd_ed25519_point_t * r,
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
make: *** [config/everything.mk:311: build/native/gcc/obj/ballet/ed25519/fd_curve25519.o] Error 1
```

--- a/src/app/fdctl/run/tiles/fd_sign.c
+++ b/src/app/fdctl/run/tiles/fd_sign.c
@@ -56,14 +56,14 @@ mux_ctx( void * scratch ) {
    we are reading incoming frags.  We don't actually need to copy the
    fragment here, see fd_dedup.c for why we do this.*/
 
-static inline void FD_FN_SENSITIVE
-during_frag( void * _ctx,
-             ulong  in_idx,
-             ulong  seq,
-             ulong  sig,
-             ulong  chunk,
-             ulong  sz,
-             int *  opt_filter ) {
+static void FD_FN_SENSITIVE
+during_frag_sensitive( void * _ctx,
+                       ulong  in_idx,
+                       ulong  seq,
+                       ulong  sig,
+                       ulong  chunk,
+                       ulong  sz,
+                       int *  opt_filter ) {
   (void)seq;
   (void)sig;
   (void)chunk;
@@ -101,16 +101,28 @@ during_frag( void * _ctx,
   }
 }
 
-static inline void FD_FN_SENSITIVE
-after_frag( void *             _ctx,
-            ulong              in_idx,
-            ulong              seq,
-            ulong *            opt_sig,
-            ulong *            opt_chunk,
-            ulong *            opt_sz,
-            ulong *            opt_tsorig,
-            int *              opt_filter,
-            fd_mux_context_t * mux ) {
+
+static void
+during_frag( void * _ctx,
+             ulong  in_idx,
+             ulong  seq,
+             ulong  sig,
+             ulong  chunk,
+             ulong  sz,
+             int *  opt_filter ) {
+  during_frag_sensitive( _ctx, in_idx, seq, sig, chunk, sz, opt_filter );
+}
+
+static void FD_FN_SENSITIVE
+after_frag_sensitive( void *             _ctx,
+                      ulong              in_idx,
+                      ulong              seq,
+                      ulong *            opt_sig,
+                      ulong *            opt_chunk,
+                      ulong *            opt_sz,
+                      ulong *            opt_tsorig,
+                      int *              opt_filter,
+                      fd_mux_context_t * mux ) {
   (void)seq;
   (void)opt_sig;
   (void)opt_chunk;
@@ -169,10 +181,23 @@ after_frag( void *             _ctx,
   ctx->out[ in_idx ].seq = fd_seq_inc( ctx->out[ in_idx ].seq, 1UL );
 }
 
+static void
+after_frag( void *             _ctx,
+            ulong              in_idx,
+            ulong              seq,
+            ulong *            opt_sig,
+            ulong *            opt_chunk,
+            ulong *            opt_sz,
+            ulong *            opt_tsorig,
+            int *              opt_filter,
+            fd_mux_context_t * mux ) {
+  after_frag_sensitive( _ctx, in_idx, seq, opt_sig, opt_chunk, opt_sz, opt_tsorig, opt_filter, mux );
+}
+
 static void FD_FN_SENSITIVE
-privileged_init( fd_topo_t *      topo,
-                 fd_topo_tile_t * tile,
-                 void *           scratch ) {
+privileged_init_sensitive( fd_topo_t *      topo,
+                           fd_topo_tile_t * tile,
+                           void *           scratch ) {
   (void)topo;
 
   FD_SCRATCH_ALLOC_INIT( l, scratch );
@@ -182,11 +207,11 @@ privileged_init( fd_topo_t *      topo,
   ctx->private_key = identity_key;
   ctx->public_key  = identity_key + 32UL;
 
-    /* The stack can be taken over and reorganized by under AddressSanitizer, 
+    /* The stack can be taken over and reorganized by under AddressSanitizer,
      which causes this code to fail.  */
 #if FD_HAS_ASAN
   FD_LOG_WARNING(( "!!! SECURITY WARNING !!! YOU ARE RUNNING THE SIGNING TILE "
-                   "WITH ADDRESS SANITIZER ENABLED. THIS CAN LEAK SENSITIVE " 
+                   "WITH ADDRESS SANITIZER ENABLED. THIS CAN LEAK SENSITIVE "
                    "DATA INCLUDING YOUR PRIVATE KEYS INTO CORE DUMPS IF THIS "
                    "PROCESS ABORTS. IT IS HIGHLY ADVISED TO NOT TO RUN IN THIS "
                    "MODE IN PRODUCTION!" ));
@@ -207,10 +232,17 @@ privileged_init( fd_topo_t *      topo,
     FD_LOG_ERR(( "madvise failed (%i-%s)", errno, fd_io_strerror( errno ) ));
 }
 
+static void
+privileged_init( fd_topo_t *      topo,
+                 fd_topo_tile_t * tile,
+                 void *           scratch ) {
+  privileged_init_sensitive( topo, tile, scratch );
+}
+
 static void FD_FN_SENSITIVE
-unprivileged_init( fd_topo_t *      topo,
-                   fd_topo_tile_t * tile,
-                   void *           scratch ) {
+unprivileged_init_sensitive( fd_topo_t *      topo,
+                             fd_topo_tile_t * tile,
+                             void *           scratch ) {
   FD_SCRATCH_ALLOC_INIT( l, scratch );
   fd_sign_ctx_t * ctx = FD_SCRATCH_ALLOC_APPEND( l, alignof( fd_sign_ctx_t ), sizeof( fd_sign_ctx_t ) );
   FD_TEST( fd_sha512_join( fd_sha512_new( ctx->sha512 ) ) );
@@ -264,6 +296,13 @@ unprivileged_init( fd_topo_t *      topo,
   ulong scratch_top = FD_SCRATCH_ALLOC_FINI( l, 1UL );
   if( FD_UNLIKELY( scratch_top > (ulong)scratch + scratch_footprint( tile ) ) )
     FD_LOG_ERR(( "scratch overflow %lu %lu %lu", scratch_top - (ulong)scratch - scratch_footprint( tile ), scratch_top, (ulong)scratch + scratch_footprint( tile ) ));
+}
+
+static void
+unprivileged_init( fd_topo_t *      topo,
+                   fd_topo_tile_t * tile,
+                   void *           scratch ) {
+  unprivileged_init_sensitive( topo, tile, scratch );
 }
 
 static ulong

--- a/src/ballet/ed25519/fd_curve25519.h
+++ b/src/ballet/ed25519/fd_curve25519.h
@@ -141,7 +141,7 @@ fd_ed25519_scalar_mul( fd_ed25519_point_t *       r,
 /* fd_ed25519_scalar_mul_base_const_time computes r = n * P, and returns r.
    n is a scalar. P is the base point.
    Note: const time implementation, safe to use with n secret. */
-fd_ed25519_point_t *
+fd_ed25519_point_t * FD_FN_SENSITIVE
 fd_ed25519_scalar_mul_base_const_time( fd_ed25519_point_t * r,
                                        uchar const          n[ 32 ] ); /* can be a secret */
 

